File: 2766-find-the-prefix-common-array-of-two-arrays/2766-find-the-prefix-common-array-of-two-arrays.py
class Solution:
    def findThePrefixCommonArray(self, A: List[int], B: List[int]) -> List[int]:
        arrLen = len(A)
        res = [0] * arrLen

        ele_A,ele_B = set(),set()

        for i in range(arrLen):
            ele_A.add(A[i])
            ele_B.add(B[i])

            count = 0
            for element in ele_A:
                if element in ele_B:
                    count += 1
            res[i] = count
        return res

File: 2766-find-the-prefix-common-array-of-two-arrays/2766-find-the-prefix-common-array-of-two-arrays.js
/**
 * @param {number[]} A
 * @param {number[]} B
 * @return {number[]}
 */
var findThePrefixCommonArray = function(A, B) {
    let len = A.length
    const res = new Array(len).fill(0)
    let ele_A = new Set()
    let ele_B = new Set()
    let count = 0

    for(let i = 0; i < len; i++){
        ele_A.add(A[i])
        ele_B.add(B[i])
        for(let element of ele_A){
            if(ele_B.has(element)){
                count++
            }
        }
        res[i] = count
        count = 0
    }
    return res
};